### DISCUSSION TITLE
Add dymaic link next to each Headings

### DISCUSSION THREAD

**Discussion by Kenzo-Wada** [withastro/starlight#2438] [Feature Requests]
### What version of `starlight` are you using?

latest

### What is your idea?

I want to automatically place an icon to the right of each Heading element that is a dynamic link to the relevant section.

### Why is this feature necessary?

Since nextra, vocs, docusaurus, and other similar starlight libraries all have this functionality

### Do you have examples of this feature in other projects?

This is what docusaurus has: 

![Screenshot from 2024-10-05 16-06-56](https://github.com/user-attachments/assets/856de094-18df-4f23-a72c-149c64e03126)

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
Thanks for the feedback :raised_hands: 

Heading anchor links is something that is being discussed in this [feature request](https://github.com/withastro/starlight/discussions/1239). I'd like to specifically point to this [comment](https://github.com/withastro/starlight/discussions/1239#discussioncomment-7775466) which contains 2 interesting links: one poiting to the the [implementation](https://github.com/withastro/docs/pull/5610) of this feature in the Astro Docs and one to a small [guide](https://hideoo.dev/notes/starlight-heading-links) I made on how to implement this feature in a Starlight project.

**Reply by Kenzo-Wada** [reply]
Okay, Ty!
